### PR TITLE
move to poetry

### LINKED ISSUE #4: ライブラリ管理をpoetryに変更

### PR DESCRIPTION
**Pull request by hirosassa**
fixes https://github.com/hotoku/bqrun/issues/4

## What I did

Move to Poetry as a package manager of this repo.
I removed `setup.py` and `setup.cfg`, and its dependent codes.

## I would like to discuss

Should we also move from Travis to GitHub Actions?

### CODE REVIEW

File: bqrun/bqrun.py
Comment (hirosassa) [inline bqrun/bqrun.py:226]: This comes from flake8's suggestion: `E741 ambiguous variable name`

File: tox.ini
Comment (hirosassa) [inline tox.ini:2]: [IMO] should remove py35 and py36 because it is going to be EOL status

https://endoflife.date/python

**Comment by hirosassa** [comment]
@hotoku plz review!

File: pyproject.toml
Comment (hirosassa) [inline pyproject.toml:14]: This plugin enables us to dynamic versioning based on tags on git repository.

**Comment by hotoku** [comment]
@hirosassa Thank you for your requests. I would like to review it but I don't have enough time right now, sorry..
But I can review until this week ends (that is 2021-12-25 or so)

**Comment by hotoku** [comment]
> Should we also move from Travis to GitHub Actions?
That's a good idea. Travis settings in this repo are automatically generated by cookiecutter and is not used, actually.

**Comment by hirosassa** [comment]
@hotoku Thanks for the quick response. 

> But I can review until this week ends (that is 2021-12-25 or so)

No worries 👍  Take your time!

> That's a good idea. Travis settings in this repo are automatically generated by cookiecutter and is not used, actually.

Sound good. So I'll create another PR to implement CI on GitHub Actions.

File: tox.ini
Comment (hotoku) [reply tox.ini:2]: Thank you for your opinion. 
Honestly speaking, I don't understand what this file exists for. It's also a automatically generated file, actually.

File: pyproject.toml
Comment (hotoku) [reply pyproject.toml:14]: Thank you for your proposal of a new tool. I have not heard about it but understand it's useful. I agree starting to use the plugin.

But before moving to poetry-dynamic-versioning, I think there needs several modifications, yet.

1. You have inserted the version placeholder for it in `pyproject.toml`. But there are another version number [here](https://github.com/hotoku/bqrun/blob/main/bqrun/__init__.py#L1).
2. The plugin's official repository's sample configuration of `build-system.requires` of `pyproject.toml` loooks `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]` but [your request](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/pyproject.toml#L34) contains only `["poetry-core>=1.0.0"]`. Is this intended? Sorry, I'm new to the plugin, so I cannot understand..

Comment (hotoku) [reply pyproject.toml:14]: @hirosassa Do you have time for the above modifications? If it's difficult, I'm OK to merge your PR as is, I will apply the modification after the merge commit.

**Review by hotoku — CHANGES_REQUESTED** [review]
I replied in your comment about poetry-dynamic-versioning, please check it.

File: pyproject.toml
Comment (hirosassa) [reply pyproject.toml:14]: Thanks for your comment.
This is from [README](https://github.com/mtkennerly/poetry-dynamic-versioning/blob/master/README.md) of the official repo.

> Poetry's typical version setting is still required in [tool.poetry], but you are encouraged to use version = "0.0.0" as a standard placeholder.

Since [tool.poetry section in pyproject.toml requires `version` settings, we have to set placeholder.

> The plugin's official repository's sample configuration of build-system.requires of pyproject.toml loooks ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]

Yeah, your suggestion is right. I didn't know that.
This change is added in [this commit](https://github.com/mtkennerly/poetry-dynamic-versioning/commit/c8e6ac0ced0ff5140166321cae1f7f0f9922fb12).

For example of [gokart repo](https://github.com/m3dev/gokart/blob/master/pyproject.toml) (I'm a maintainer of the repo), we didn't write `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]`.
But this change is not tested, so I'll take a look.

Comment (hotoku) [reply pyproject.toml:14]: About 1. of my comment.

Sorry, my explanation was ambiguous and my point was missed.

In this package, there are 2 places where the version number is referred.

1. in `/pyproject.toml`: `[tool.poetry] version = ...` [here](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/pyproject.toml#L3)
2. in `/bqrun/__init__.py`: `__version__ = ...` [here](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/bqrun/__init__.py#L1)

In your PR, the former is written as placeholder ("0.0.0"), but the latter is left as the concrete number ("1.3.1").

To my understand, the latter should be written as placeholder

Comment (hirosassa) [reply pyproject.toml:14]: Oh, it's my misunderstanding. 
> in `/bqrun/__init__.py`: `__version__ = ...` 

This should be removed.

File: tox.ini
Comment (hirosassa) [reply tox.ini:2]: This is `tox`'s test environment flag. If you add `pyxy` here, the tests run on python x.y version.
This feature is useful for library development (you can check your library correctly run on several python versions), but it is not necessary for this repo because this is CLI tool.
So my recommendation is omitting `py35-38` to reduce CI running time on GitHub Actions (I'll create it in next PR).

File: pyproject.toml
Comment (hirosassa) [reply pyproject.toml:14]: @hotoku I tested my local env and found it that `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]` this entry is not necessary, but it is described on the README on official repo, so I added in 6413a73.

Also, I removed version setting on `__init__.py` in 99f6b53

**Review by hotoku — APPROVED** [review]

**Comment by hotoku** [comment]
@hirosassa thank you. I merged it.

**Pull request merged**
